### ISSUE TITLE
Error in Apiary reporter

### ISSUE THREAD

**Issue by netmilk** [apiaryio/dredd#289]
Under some circumstances, Apiary reporter throws an exception. Unfortunately I have no more info about it: 

```
info: Beginning Dredd testing...
fail: GET / duration: 123ms
fail: GET /tenants duration: 153ms
fail: GET /tenants/1001/canvases duration: 119ms
error: POST /tenants/1001/canvases duration: 65ms
error: Error: Parse Error
at Socket.socketOnData (http.js:1584:20)
at TCP.onread (net.js:528:27)

/usr/lib/node_modules/dredd/lib/reporters/apiary-reporter.js:160
if ((base = data['resultData']['result'])['general'] == null) {
^
TypeError: Cannot read property 'general' of undefined
at EventEmitter.<anonymous> (/usr/lib/node_modules/dredd/lib/reporters/apiary-reporter.js:160:50)
at EventEmitter.emit (events.js:117:20)
at ClientRequest.<anonymous> (/usr/lib/node_modules/dredd/lib/transaction-runner.js:517:33)
at ClientRequest.emit (events.js:95:17)
at Socket.socketOnData (http.js:1589:9)
at TCP.onread (net.js:528:27)
```

**Comment by romansklenar** [comment]
I've run into same issue using Apiary reporter.

**Comment by netmilk** [comment]
@romansklenar Thanks for reporting this! Would you mind providing some more information about your system and version of Dredd you're using?

**Comment by masone** [comment]
```
❯ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.11.2
BuildVersion:	15C50

❯ npm -v
3.5.1
                                                                                                                                                           
❯ node -v
v4.2.3

❯ rails -v
Rails 5.0.0.beta1

❯ dredd --version
dredd v1.0.2
```

I get the same error as @netmilk. Running a rails 5 beta server powered by puma.
I just started with apiary, so my blueprint is minimal.

**Comment by netmilk** [comment]
@masone may you please share the whole Dredd command and/or `dredd.yml` file, please?

**Comment by masone** [comment]
Sure, thanks for looking into this Adam! Here's a gist: https://gist.github.com/masone/a9d29cd314388ccaffbf

**Comment by masone** [comment]
The config is generated by the cli tool. In case you're wondering about hooks: I haven't set up any. Just installed the `dredd_hooks` gem.

**Comment by netmilk** [comment]
@masone Thanks for additional info! BRB :)

**Comment by netmilk** [comment]
@masone Thank you again for the super helpful debug info you submitted here. I manage to hunt this bug down and here is a PR with fix #331. I'll do the release after PR review and merge. wink-wink @honzajavorek @w-vi

**Comment by masone** [comment]
:tada: thanks Adam!

**Comment by adriaanbalt** [comment]
+1

**Comment by adriaanbalt** [comment]
started to work all the time now.

**Comment by mikz** [comment]
Installed latest version from npm and have the same issue. Also rails 5 + puma stack.

**Comment by honzajavorek** [comment]
I managed to get this error when testing the reporter while having the server down. Just noting this so we could write tests for that scenario.

Initial implementation of the fix is in #331, we're soon about to get to it and finish it, but #332 had to go first.
